app/models/spot.rb

```diff
@@ -48,7 +48,24 @@ def create_matches
                            .where.not(user_id: user_id)
 
     compatible_spots.each do |spot|
-      matches.create!(first_user_id: user_id, second_user_id: spot.user_id)
+      create_match(spot)
     end
   end
+
+  def create_match(spot)
+    match = matches.create!(first_user_id: user_id, second_user_id: spot.user_id)
+    notify(match, match.second_user)
+  end
+
+  def notify(match, matched_user)
+    data = {
+      name: matched_user.name,
+      avatar: matched_user.avatar.url,
+      match_id: match.id
+    }
+
+    return unless matched_user.push_token
```

Comment:
You should move this to the beginning because if the `push_token` is not present,  you will not use the `data`